File: codegen/codegen-erlang/src/test/resources/ir/rest_json_encode_get_name_request.expected.erl
%% @doc Encode HTTP request for smithy.beam.demo.http#GetName.
-spec encode_get_name_request(get_name_input()) -> #http_request{}.
encode_get_name_request(Input = #get_name_input{name = Name}) ->
    Path = <<"/names/", (uri_encode(to_binary(Name)))/binary>>,
    Query = [],
    Headers = [{<<"Content-Type">>, <<"application/json">>}],
    Body = <<>>,
    #http_request{
        method = <<"GET">>,
        path = Path,
        query = maps:from_list(Query),
        headers = Headers,
        body = Body
    }.

File: codegen/codegen-elixir/src/test/resources/ir/rest_json_encode_get_name_request.expected.ex
@spec encode_get_name_request(HttpServiceTypes.GetNameInput.t()) :: %RuntimeTypes.HttpRequest{}
def encode_get_name_request(input) do
  path = "/names/" <> uri_encode(input.name)
  query = %{}
  headers = [{"Content-Type", "application/json"}]
  body = ""
  %RuntimeTypes.HttpRequest{
    method: "GET",
    path: path,
    query: query,
    headers: headers,
    body: body
  }
end